pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -209,12 +214,17 @@ type BuildStatus struct {
 
 	// StepStates describes the state of each build step container.
 	StepStates []corev1.ContainerState `json:"stepStates,omitEmpty"`
-	// Conditions describes the set of conditions of this build.
-	Conditions []BuildCondition `json:"conditions,omitempty"`
+
 	// StepsCompleted lists the name of build steps completed.
 	StepsCompleted []string `json:"stepsCompleted"`
+
+	// Conditions describes the set of conditions of this build.
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty"`
```

Comment:
you can update this to the following:

```
Conditions duck.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Reply:
This is only going to be meaningful for API servers.